server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
-		return websocket.BinaryMessage, protoUnmarshalErr
+		return "", protoUnmarshalErr
 	}
 	// Create a [32]byte{} identityPublicKey to satisfy ed25519.Verify() type requirements
 	identityPublicKey := [32]byte{}
 	// Create a [64]byte{} signedByteSequenceFromClient to satisfy ed25519.Verify() type requirements
 	signedByteSequenceFromClient := [64]byte{}
+	// Create a string representation of the Identity Public Key
+	identityPublicKeyString := string(messageFromClient.Response.Auth.IdentityPublicKey)
 	// Decode the hex representation of the identityPublicKey here as transmitting it over the network without hex encoding can cause issues with some clients
-	identityPublicKeyDecoded, identityPublicKeyDecodedErr := hex.DecodeString(string(protobufAuth.GetIdentityPublicKey()))
+	identityPublicKeyDecoded, identityPublicKeyDecodedErr := hex.DecodeString(identityPublicKeyString)
 	if identityPublicKeyDecodedErr != nil {
 		// If there is an error while decoding the hex identity public key, terminate the connection
-		return websocket.BinaryMessage, identityPublicKeyDecodedErr
+		return "", identityPublicKeyDecodedErr
+	}
+	// Get the byte sequence which was signed by the client
+	byteSequenceThatClientSigned := messageFromClient.Response.Auth.ToSign
+	// Check if the byte sequence that was signed by the client contains the initial bytes we sent to the client
+	if !bytes.HasPrefix(byteSequenceThatClientSigned, backendRandomBytes) {
+		// If the client has modified the bytes we sent, return an error pointing out that this behavior is not allowed
+		return "", errors.New("Client is only allowed to append a byte sequence, and not to modify the one which was sent")
 	}
 	// Fill the newly created identityPublicKey with the hex decoded representation of the IdentityPublicKey contained in the response from the client
 	copy(identityPublicKey[:], identityPublicKeyDecoded)
 	// Fill the newly created signedByteSequenceFromClient with the Signature contained in the response from the client
-	copy(signedByteSequenceFromClient[:], protobufAuth.GetSignature())
+	copy(signedByteSequenceFromClient[:], messageFromClient.Response.Auth.Signature)
 	// Verify the validity of the signature using ed25519.Verify()
-	if ed25519.Verify(&identityPublicKey, byteSequenceToSign, &signedByteSequenceFromClient) {
-		// If the signed byte sequence from client has a valid signature, change the websocket connection state and keep the websocket connection alive
-		return websocket.BinaryMessage, nil
+	if ed25519.Verify(identityPublicKey[:], byteSequenceThatClientSigned, signedByteSequenceFromClient[:]) {
+		// If the signed byte sequence from client has a valid signature, echo the authentication attempt back to the client so that he knows it was successful
+		writeMessageError := websocketConnection.WriteMessage(websocket.BinaryMessage, messageFromClientProto)
+		// If there is an error while sending a message to a client
+		if writeMessageError != nil {
```

Comment:
From my point of view we don't need a comment here since it's a simple condition. But thats just my opinion, you can leave it if you disagree :)

Reply:
Agreed. Will aim to omit commenting on simple and already descriptive conditions.